Source/Painting/SvgLinearGradientServer.cs

```diff
@@ -78,44 +69,37 @@ public override Brush GetBrush(SvgVisualElement renderingElement, ISvgRenderer r
 
                 using (var transform = EffectiveGradientTransform)
                 {
-                    var midPoint = new PointF((points[0].X + points[1].X) / 2, (points[0].Y + points[1].Y) / 2);
                     transform.Translate(bounds.X, bounds.Y, MatrixOrder.Prepend);
                     if (this.GradientUnits == SvgCoordinateUnits.ObjectBoundingBox)
                     {
                         // Transform a normal (i.e. perpendicular line) according to the transform
                         transform.Scale(bounds.Width, bounds.Height, MatrixOrder.Prepend);
-                        transform.RotateAt(-90.0f, midPoint, MatrixOrder.Prepend);
                     }
                     transform.TransformPoints(points);
                 }
 
+                points[0].X = (float)Math.Round(points[0].X, 4);

```

Comment:
Unnecessary rotations have been deleted, but there are user-specified rotations, so round off to eliminate errors.